lib/stripe/subscriptions/usage.ex

```diff
@@ -0,0 +1,84 @@
+defmodule Stripe.SubscriptionItem.Usage do
+  @moduledoc """
+  Work with Stripe usage record objects.
+
+  Stripe API reference: https://stripe.com/docs/api/usage_records
+  """
+
+  use Stripe.Entity
+  import Stripe.Request
+
+  @type t :: %__MODULE__{
+          id: Stripe.id(),
+          object: String.t(),
+          livemode: boolean,
+          quantity: non_neg_integer,
+          subscription_item: Stripe.id() | Stripe.SubscriptionItem.t() | nil,
+          timestamp: Stripe.timestamp() | nil
+        }
+
+  defstruct [
+    :id,
+    :object,
+    :livemode,
+    :quantity,
+    :subscription_item,
+    :timestamp
+  ]
+
+  @plural_endpoint "subscription_items"
+
+  @doc """
+  Creates a usage record for a specified subscription item and date, and fills it with a quantity.
+  """
+  @spec create(params, Stripe.options()) :: {:ok, t} | {:error, Stripe.Error.t()}
+        when params: %{
+                :quantity => float,
+                :subscription_item => Stripe.id() | Stripe.SubscriptionItem.t(),
+                :timestamp => Stripe.timestamp() | non_neg_integer,
+                optional(:action) => String.t()
+             } 
+  def create(params, opts \\ []) do
+    url = 
+    params 
+    |> Map.pop(:subscription_item)
+    |> build_url
+
+    new_request(opts)
+    |> put_endpoint(url)
+    |> put_method(:post)
+    |> put_params(params)
```

Comment:
+ `|> cast_to_id([:subscription_item])`

Reply:
Do you want this before the `put_params` or where in the flow? (Was going to guess after, but would rather ask).

Reply:
(dup of comment above) We need |> put_param(:subscription_item, id) since that is a required param to pass to stripe.

Reply:
From their docs, it looks like :subscription_item is required, but as a required path param. When passing it along with the other request parameters, Stripe returns an error that the additional properties are not allowed: "subscription item"

```
{:error,
             %Stripe.Error{
               code: :invalid_request_error,
               extra: %{
                 http_status: 400,
                 raw_error: %{
                   "message" => "Request validation error: validator 0xc00049ba70 failed: additional properties are not allowed: subscription_item",
                   "type" => "invalid_request_error"
                 }
               },
               message: "Request validation error: validator 0xc00049ba70 failed: additional properties are not allowed: subscription_item",
               request_id: {"Request-Id", "req_123"},
               source: :stripe,
               user_message: nil
             }}
```

Reply:
Oh you are right!  THx for testing